core/unix/memquery.c

```diff
@@ -40,16 +40,24 @@
 #include "memquery.h"
 #include "module.h"
 
+#if defined(STANDALONE_UNIT_TEST)
+#    include "memquery_test.h"
+#else
+#    define MEMQUERY_ITERATOR_START memquery_iterator_start
```

Comment:
I like the idea of using `#define memquery_iterator_start ...` and making the code look more normal; as you mention it's more direct: it also saves a click when looking through the code with Kythe (or equivalent). My only concern would be that it might not be as obvious to editors that the code has assumptions about which functions are being swapped out, but if we just document the macro trickiness in a comment in that file I believe that will alleviate any future confusion.

I considered option B earlier, but I guess I didn't think it was much better than option A until I started typing this reply. I wanted to make callers still look the same, but a simple way to make this a no-op for callers is to replace the current `memquery_library_bounds_by_iterator` with `memquery_library_bounds_by_iterator_with_funcs` (or some better name) and write a new `memquery_library_bounds_by_iterator` which calls `memquery_library_bounds_by_iterator_with_funcs` with the "real" dependency logic. Then the unit test can just call `memquery_library_bounds_by_iterator_with_funcs` with its fake functions directly, allowing us to sidestep the hacky `enable_memquery_unit_test`. It's certainly much more encapsulated that way, approximating option C, and avoids the confusion about what functions are being faked out like lowercase-option-A does.

(I also considered something like option C but I decided to avoid sinking so much time into arguing with cmake; needless to say I was pretty happy when I realized the unit_tests target worked just fine for this.)

So sounds like we want option B. I'll see how this decision jives with the other comments you've left and let you know when I'm ready for you to take another look.

Reply:
Since it's a smaller change from what you have and thus less work if you want to do A with lowercase I am fine with that: just want to say don't spend time on B just because of me, only if you agree it's nicer.

Reply:
B is definitely nicer since it lets us remove `enable_memquery_unit_test`, and it's fairly simple to integrate into the code.